File: src/ingest.py
import json
import boto3
import os

sqs = boto3.client('sqs')
# Esta variable la tomará automáticamente de tu configuración de Terraform
QUEUE_URL = os.environ.get('QUEUE_URL')

def handler(event, context):
    for record in event['Records']:
        bucket = record['s3']['bucket']['name']
        key = record['s3']['object']['key']
        
        message = {
            'bucket': bucket,
            'key': key,
            'status': 'uploaded'
        }
        
        sqs.send_message(
            QueueUrl=QUEUE_URL,
            MessageBody=json.dumps(message)
        )
        
    return {
        'statusCode': 200,
        'body': json.dumps('Notificación enviada a SQS')
    }

File: iac/src/ingest.js
const { SQSClient, SendMessageCommand } = require("@aws-sdk/client-sqs");
const sqs = new SQSClient({});

exports.handler = async (event) => {
    const queueUrl = process.env.QUEUE_URL; // Asegúrate de que coincida con tu variable de entorno en Terraform

    for (const record of event.Records) {
        const message = {
            bucket: record.s3.bucket.name,
            key: record.s3.object.key,
            status: 'uploaded'
        };

        const command = new SendMessageCommand({
            QueueUrl: queueUrl,
            MessageBody: JSON.stringify(message),
        });

        try {
            await sqs.send(command);
            console.log("Notificación enviada a SQS:", message);
        } catch (err) {
            console.error("Error enviando a SQS:", err);
        }
    }

    return {
        statusCode: 200,
        body: JSON.stringify('Notificación procesada exitosamente')
    };
};